lib/mayaUsd/fileio/jobs/meshDataReadJob.h

```diff
@@ -0,0 +1,62 @@
+//
+// Copyright 2022 Autodesk
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#ifndef PXRUSDMAYA_MESHDATAREAD_JOB_H
+#define PXRUSDMAYA_MESHDATAREAD_JOB_H
+
+#include <mayaUsd/fileio/jobs/readJob.h>
+
+#include <maya/MObject.h>
+
+#include <vector>
+
+PXR_NAMESPACE_OPEN_SCOPE
+
+// Custom class to prevent read job from creating new nodes and to access
+// geometric data directly.
+class UsdMaya_MeshDataReadJob : public UsdMaya_ReadJob
+{
+public:
+    MAYAUSD_CORE_PUBLIC
+    UsdMaya_MeshDataReadJob(
+        const MayaUsd::ImportData&  iImportData,
+        const UsdMayaJobImportArgs& iArgs);
+
+protected:
+    // Override prim reader to create a mesh without a node and to store
+    // the mesh then access it and return it in the plug.
```

Comment:
The comment talks about "the plug" but there are no plug in sight. I assume it is talking about some plug of the caller? It would help the reader to c;larify this.